tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs

```diff
@@ -205,6 +214,19 @@ private async Task RestorePinnedWidgetsAsync()
         }
     }
 
+    private async Task DeleteAbandonedWidget(Widget widget)
+    {
+        var length = ViewModel.WidgetHostingService.GetWidgetHost()?.GetWidgets().Length;

```

Comment:
what is the value for length if the return value of GetWidgetHost is null?

Reply:
Should probably be a ! instead of a ?. Something catastrophic would have had to happen for it to get here and return null after getting past
`var hostWidgets = ViewModel.WidgetHostingService.GetWidgetHost()?.GetWidgets();`
and the null check at the top of `RestorePinnedWidgetsAsync`.